tests/ert/unit_tests/scheduler/test_slurm_driver.py

```diff
@@ -173,13 +173,51 @@ async def test_project_code_is_set(project_code):
 
 
 @pytest.mark.usefixtures("capturing_sbatch")
-@given(max_runtime=st.integers(min_value=1))
-async def test_max_runtime_is_set(max_runtime):
-    driver = SlurmDriver(max_runtime=str(max_runtime))
+@given(max_runtime=st.floats(min_value=1, max_value=999999999))
+async def test_max_runtime_is_properly_formatted(max_runtime):
+    # According to https://slurm.schedmd.com/sbatch.html we accept the formats
+    # "minutes", "minutes:seconds", "hours:minutes:seconds", "days-hours",
+    # "days-hours:minutes" and "days-hours:minutes:seconds".
+    driver = SlurmDriver(max_runtime=max_runtime)
     await driver.submit(0, "sleep", name="myjobname")
-    assert f"--time={max_runtime}" in Path("captured_sbatch_args").read_text(
-        encoding="utf-8"
+    cmd_args = Path("captured_sbatch_args").read_text(encoding="utf-8").split()
+    time_argument = next(
+        arg.split("=")[1] for arg in cmd_args if arg.startswith("--time")
     )
+    if "-" in time_argument:
+        int(time_argument.split("-")[0])
+        hhmmss = time_argument.split("-")[1]
+    else:
+        hhmmss = time_argument
+    hh, mm, ss = map(int, hhmmss.split(":"))
+    assert 0 <= hh < 24
+    assert 0 <= mm < 60
+    assert 0 <= ss < 60
+
+
+@pytest.mark.usefixtures("capturing_sbatch")
+@pytest.mark.parametrize(
+    "seconds, expected_format",
+    [
+        (0, ""),
+        (0.1, ""),
+        (1, "0:00:01"),
+        (1.9, "0:00:01"),
+        (61, "0:01:01"),
+        (60 * 60, "1:00:00"),
+        (24 * 60 * 60 - 1, "23:59:59"),
+        (24 * 60 * 60, "1-0:00:00"),
+    ],
+)
+async def test_max_runtime_formatting_samples(seconds, expected_format):
```

Comment:
As this is part of unit test, isn't it enough to write a test for  `def _seconds_to_slurm_time_format(seconds: float) -> str:`?

Reply:
Split the test function in two, check the fixup commit.